docs/reference/online-stores/elasticsearch.md

```diff
@@ -0,0 +1,120 @@
+# Elasticsearch online store
+
+## Description
+
+The [Elasticsearch](https://www.elastic.co/) online store provides support for materializing feature values into an Elasticsearch database for serving online features.
+
+Features:
+* Creating indexes and documents through parameters in FeatureView
+* Support for float and binary vector embeddings
+* Full support for HNSW indexing for available metric types in Elasticsearch
+
+## Getting started
+In order to use this online store, you'll need to install the elasticsearch extra:
+- `pip install 'feast[elasticsearch]'`
+
+## Example
+
+Connecting to an Elasticsearch instance:
+
+{% code title="feature_store.yaml" %}
+```yaml
+project: my_feature_repo
+registry: data/registry.db
+provider: local
+online_store:
+  type: elasticsearch
+  endpoint: ELASTICSEARCH_URL
+  username: ELASTICSEARCH_USERNAME
+  password: ELASTICSEARCH_PASSWORD
+```
+{% endcode %}
+
+The full set of configuration options is available in [ElasticsearchOnlineStoreConfig](https://github.com/ExpediaGroup/feast/blob/master/sdk/python/feast/expediagroup/vectordb/elasticsearch_online_store.py#L40).
+
+## Elasticsearch Online Store Format
+
+### Overview
+
+Elasticsearch is a document search engine with vector embedding indexing and knn support.
+
+Elasticsearch organizes data into _indexes_. An index contains a series of object documents. A document can contain one vector field one the top-level of a document. A vector can be either an array of floats or binary (bytes). The vector is indexed using Hierarchical Navigable Small World (HNSW) graph.
```

Comment:
change `can contain one vector field one the top-level` to `can contain one vector field on the top-level`

Reply:
I feel it might be a good idea explaining that a vector index is not the same as an Elasticsearch index. I won't be surprised if people get that mixed up